ODC-7383: add support for tektonresult api to load TaskRuns

vikram-raj opened the pull request:
Story: https://issues.redhat.com/browse/ODC-7383

Descriptions:

Add support to fetch TaskRuns using the TektonResults API endpoint in the TaskRun list page, TaskRuns list on the PipelineRun details page.


Test setup:
1. Install Red Hat Openshift Pipeline Operator in the console
2. Install Tekton Results in the console using this gist https://gist.github.com/vikram-raj/257d672a38eb2159b0368eaed8f8970a
3. Create Pipeline and PipelineRuns and try out all the features.

vikram-raj commented:
/cc @jerolimov @karthikjeeyar 

/uncc @andrewballantyne @abhinandan13jan

vikram-raj commented:
/retest

lokanandaprabhu commented on frontend/packages/pipelines-plugin/src/components/pipelineruns/hooks/useTektonResults.ts:
```diff
@@ -0,0 +1,199 @@
+    resultTaskRunsLoaded,
+    resultTaskRunsLoadError,
+    getNextPage,
+  ] = useTRTaskRuns(ns);
```
Here you are passing only the namespace but not options to select for particular pipelineRun.

lokanandaprabhu commented on frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/PipelineRuns.tsx:
```diff
@@ -1,7 +1,7 @@
 import { useTranslation } from 'react-i18next';
-import { ListPage } from '@console/internal/components/factory';
+// import { ListPage } from '@console/internal/components/factory';
 import { RowFilter } from '@console/internal/components/filter-toolbar';
```
Can you remove this commented line.

karthikjeeyar commented:
I can see infinite calls to tekton-results API in pipelineRuns -> taskRuns tab

![image](https://github.com/openshift/console/assets/9964343/d67a501b-fe47-4c9f-b846-bc18862e2ff5)

karthikjeeyar commented:
I think Search page -> TaskRuns list is using default resource list component so the tekton-results call is missing here, do we want to use the same `TaskRunList` component (with tekton-results) for search page as well? 



![image](https://github.com/openshift/console/assets/9964343/b914bcf1-926b-43a9-ac55-82af8105be18)

karthikjeeyar commented:
Search Page -> PipelineRun list makes tekton-results to API call but the data is not rendered in the list view.

![image](https://github.com/openshift/console/assets/9964343/cf671c1c-8792-44a8-a03c-089a38d96d9a)

vikram-raj commented:
@karthikjeeyar Regarding the search page. That is expected. We are not showing the data from the result database in the search page list as on the search page we only show k8s data.

vikram-raj commented:
I fixed the infinite network call and added changes to @lokanandaprabhu PR https://github.com/openshift/console/pull/13328 in the commit  https://github.com/openshift/console/pull/13328/commits/7c77c26ceb284ec83d2c5f7ba4831f70aa4e1598 as we shared the utils for both the PR.

vikram-raj commented:
/close 

We merge these changes in the PR https://github.com/openshift/console/pull/13328
